test/integration/collection-view.spec.js

```diff
@@ -0,0 +1,139 @@
+const _ = Cypress._;
+const $ = Cypress.$;
+
+describe('CollectionView', () => {
+  let BbModel;
+  let BbCollection;
+  let CollectionView;
+  let MnView;
+
+  beforeEach(() => {
+    cy.server();
+
+    cy.visitMn(function({Marionette, Backbone}) {
+      BbModel = Backbone.Model;
+      BbCollection = Backbone.Collection;
+      MnView = Marionette.View;
+
+      const FooView = Marionette.View.extend({
+        template: _.template('Foo template content')
+      });
+
+      const BarView = Marionette.View.extend({
+        template: _.template('Bar template content')
+      });
+
+      CollectionView = Marionette.CollectionView.extend({
+        className: 'collection-view-container',
+        collection: new BbCollection(),
+        childView(item) {
+          if (item.get('isFoo')) {
+            return FooView;
+          } else {
+            return BarView;
+          }
+        }
+      });
+
+    });
+  });
+
+  describe('childView', () => {
+    let collectionView;
+    let fooModel;
+    let barModel;
+
+    beforeEach(() => {
+      const Model = BbModel.extend({
+        defaults: {
+          isFoo: false
+        }
+      });
+
+      fooModel = new Model({
+        isFoo: true
+      });
+
+      barModel = new Model({
+        isFoo: false
+      });
+
+      collectionView = new CollectionView();
+      collectionView.render();
+
+      $('#app-hook').html(collectionView.$el);
+    });
+
+    it('should be without childView\'s', function() {
+      cy.get('.collection-view-container').should('be.empty');
+    });
+
+    it('should show default childView', function() {
+      collectionView.collection.add(fooModel);
+
+      cy.get('#app-hook').should('contain', 'Foo template content');
```

Comment:
I think `#app-hook` should be set as an alias
[`cy.get("#app-hook").as("app-hook")`](https://docs.cypress.io/docs/using-aliases#section-aliasing-dom-elements) for `cy.get("@app-hook")` usage instead of re-querying the document. todo for now